pkg/controller/plan/kubevirt.go

```diff
@@ -935,14 +936,27 @@ func (r *KubeVirt) GetGuestConversionPod(vm *plan.VMStatus) (pod *core.Pod, err
 	return
 }
 
-func (r *KubeVirt) UpdateVmByConvertedConfig(vm *plan.VMStatus, pod *core.Pod, step *plan.Step) (err error) {
+func (r *KubeVirt) getInspectionXml(pod *core.Pod) (string, error) {
+	inspectionUrl := fmt.Sprintf("http://%s:8080/inspection", pod.Status.PodIP)
+	resp, err := http.Get(inspectionUrl)
+	if err != nil {
+		return "", liberr.Wrap(err)
+	}
+	inspectionBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", liberr.Wrap(err)
+	}
+	defer resp.Body.Close()
+	return string(inspectionBytes), nil
+}
+
+func (r *KubeVirt) UpdateVmByConvertedConfig(vm *plan.VMStatus, pod *core.Pod, step *plan.Step) error {
 	if pod.Status.PodIP == "" {
```

Comment:
I know this is not part of this PR, but
pod is a pointer and can theoretically be nil, we need check for pod is nil, before using it ...